gluten-ut/spark32/src/test/scala/org/apache/spark/sql/execution/joins/GlutenBroadcastJoinSuite.scala

```diff
@@ -136,21 +145,21 @@ class GlutenBroadcastJoinSuite extends BroadcastJoinSuite with GlutenTestsCommon
       /* ######## test cases for non-equal join ######### */
       withSQLConf(SQLConf.CROSS_JOINS_ENABLED.key -> "true") {
         // INNER JOIN && t1Size < t2Size => BuildLeft
-        assertJoinBuildSide("SELECT /*+ MAPJOIN(t1, t2) */ * FROM t1 JOIN t2", bl, BuildLeft)
+        assertJoinBuildSide("SELECT /*+ MAPJOIN(t1, t2) */ * FROM t1 JOIN t2", blt, BuildLeft)
         // FULL JOIN && t1Size < t2Size => BuildLeft
         assertJoinBuildSide("SELECT /*+ MAPJOIN(t1, t2) */ * FROM t1 FULL JOIN t2", bl, BuildLeft)
         // FULL OUTER && t1Size < t2Size => BuildLeft
         assertJoinBuildSide("SELECT * FROM t1 FULL OUTER JOIN t2", bl, BuildLeft)
         // LEFT JOIN => BuildRight
-        assertJoinBuildSide("SELECT /*+ MAPJOIN(t1, t2) */ * FROM t1 LEFT JOIN t2", bl, BuildRight)
+        assertJoinBuildSide("SELECT /*+ MAPJOIN(t1, t2) */ * FROM t1 LEFT JOIN t2", blt, BuildRight)
         // RIGHT JOIN => BuildLeft
-        assertJoinBuildSide("SELECT /*+ MAPJOIN(t1, t2) */ * FROM t1 RIGHT JOIN t2", bl, BuildLeft)
+        assertJoinBuildSide("SELECT /*+ MAPJOIN(t1, t2) */ * FROM t1 RIGHT JOIN t2", blt, BuildLeft)
 
         /* #### test with broadcast hint #### */
         // INNER JOIN && broadcast(t1) => BuildLeft
-        assertJoinBuildSide("SELECT /*+ MAPJOIN(t1) */ * FROM t1 JOIN t2", bl, BuildLeft)
+        assertJoinBuildSide("SELECT /*+ MAPJOIN(t1) */ * FROM t1 JOIN t2", blt, BuildLeft)
         // INNER JOIN && broadcast(t2) => BuildRight
-        assertJoinBuildSide("SELECT /*+ MAPJOIN(t2) */ * FROM t1 JOIN t2", bl, BuildRight)
+        assertJoinBuildSide("SELECT /*+ MAPJOIN(t2) */ * FROM t1 JOIN t2", blt, BuildRight)
```

Comment:
Do the statements checks on physical plan only? Do we need test with result-checking to cover the added code?

I see we had enabled `Gluten...JoinSuite`s but I am not sure if the test cases works for Gluten's BNLJ either. Would you like to help check that? Thanks.

Reply:
`GlutenInnerJoinSuite` and `GlutenOuterJoinSuite` are enabled in the Gluten. Both of these suites check the answer of test cases. In these test cases, explicit joins are selected and since now we have added `BNLJTransformer`, BNLJ is getting transformed into Gluten for `inner`, `leftOuter with BuildRight` and `RightOuter with BuildLeft` joins.

Reply:
Here in `GlutenBroadcastJoinSuite`, it is not checking the answer and only checking physical plan. This test case is a rewrite of spark test case in Gluten and I had to make changes in to get it passed in pipeline.

please let me know if I should add more test cases.

The few testcases I can think of is to add fallback test cases for other remaining join types i.e. `FullOuter`, `LeftOuter with BuildLeft` & `RightOuter with BuildRight`.

Reply:
Thanks for your explanation. If `GlutenBroadcastJoinSuite` mainly checks against plan then we don't have to add answer-checking code in this suite. I was just worried whether we actually have at least some code to verify the correctness of the patch's code. If `GlutenInnerJoinSuite` `GlutenOuterJoinSuite` do work for BNLJ transformer, then it's fine enough to me.

> The few testcases I can think of is to add fallback test cases for other remaining join types i.e. FullOuter, LeftOuter with BuildLeft & RightOuter with BuildRight.

That's great idea. You can decide whether to put code in this PR or in a new one. Thanks.